### PR TITLE
persist: hook maelstrom up to ci

### PR DESCRIPTION
**Pull request by danhhz**
### Motivation

  * This PR adds a feature that has not yet been specified.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: test/persist/mzcompose.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+SERVICES = [
+    Service(
+        "maelstrom-persist",
+        {"mzbuild": "maelstrom-persist", "volumes": ["./maelstrom:/store"]},
+    ),
+]
+
+
+def workflow_default(c: Composition) -> None:
```
Comment (danhhz) [inline test/persist/mzcompose.py:21]: this is needlessly hardcoded, but I'm mostly just looking to get this hooked up to CI without spending too much time on it. for some reason qemu seems to be more unhappy about running things locally than the last time I worked on this, so iteration is hard

File: src/persist-client/ci/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    git \
+    gnuplot \
+    openjdk-11-jre # WIP is this right?
+
+RUN curl -fsSL https://github.com/jepsen-io/maelstrom/releases/download/v0.2.1/maelstrom.tar.bz2 \
+    | tar -xj --strip-components=2 maelstrom/lib/maelstrom.jar
+
+COPY examples/persistcli /usr/local/bin/persistcli
+
+ENTRYPOINT ["java", "-Djava.awt.headless=true", "-jar", "/maelstrom.jar", "test", "-w", "txn-list-append", "--bin=/usr/local/bin/persistcli"]
```
Comment (danhhz) [inline src/persist-client/ci/Dockerfile:24]: any opinion on where we put the jar?

File: src/persist-client/ci/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# by the Apache License, Version 2.0.
+
+MZFROM ubuntu-base
+
+RUN apt-get update && TZ=UTC DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    bzip2 \
+    curl \
+    git \
+    gnuplot \
+    openjdk-11-jre # WIP is this right?
```
Comment (danhhz) [inline src/persist-client/ci/Dockerfile:17]: I can't use openjdk-8-jre anymore, apparently. the 0.2.1 release of maelstrom is compiled with something too new. any opinions on openjdk vs I dunno something else?

**Review by danhhz — COMMENTED** [review]
@benesch I finally got a chance to circle back to this! didn't spend much time cleaning it up because I don't really have opinions on any of this and I figured you might

File: src/persist-client/ci/mzbuild.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+name: maelstrom-persist
+pre-image:
+  - type: cargo-build
+    example: persistcli
+    strip: false
+publish: false
```
Comment (danhhz) [inline src/persist-client/ci/mzbuild.yml:15]: without this, I get an error in the build job. but with it, the mzcompose invocation seems to cargo build the persistcli  example. advice?

https://buildkite.com/materialize/tests/builds/35920#01814fb4-b269-4969-a952-ad5f82521674

Comment (benesch) [reply src/persist-client/ci/mzbuild.yml]: Ah yeah, we have a funny setup where I have to go into Docker Hub and pre-create the repository for `publish: true`. Fixed now!

File: src/persist-client/ci/Dockerfile (hunk 3 above)
Comment (benesch) [reply src/persist-client/ci/Dockerfile]: openjdk-11-jre is exactly right!

File: src/persist-client/ci/Dockerfile (hunk 2 above)
Comment (benesch) [reply src/persist-client/ci/Dockerfile]: Oh, /usr/local/share/java, I suppose!

**Comment by benesch** [comment]
Sweet, seems to work! I resolved all your questions and marked for automerge!

**Comment by danhhz** [comment]
TFTR and the fixes!

**Pull request merged**